src/rust/ide/src/ide/integration.rs

```diff
@@ -988,6 +1005,11 @@ impl Model {
 
         debug!(self.logger, "Attaching visualization on {node_id}.");
         let visualization  = self.prepare_visualization(node_id)?;
+        let analytics_event = analytics::data::Public::new("visualization_enabled_in_ui");
+        let analytics_data = format!("{:?}", visualization);
+        let analytics_data_public = analytics::data::Public::new(&analytics_data);
```

Comment:
this should be implemented as `analytics::data::Public(|| format!("{:?}", visualization))`, sothis will not be run when analitics is off